tools/docker/Dockerfile

```diff
@@ -0,0 +1,184 @@
+# syntax=docker/dockerfile:experimental
+
+# Common tools needed for the build stages ahead. The final end-use images (that we run tests in) are not derived from
+# this image, so the installs here will not take up space in those final images.
+FROM debian:stretch as build-base
+ENV PATH /bin:/usr/bin:/usr/local/bin:/sbin:/usr/sbin:/usr/local/sbin:/usr/lib/go/bin:/root/.cargo/bin
+RUN --mount=type=cache,id=aptcache,target=/var/cache/apt --mount=type=cache,id=aptlib,target=/var/lib/apt \
+mkdir -p /etc/apt/sources.list.d \
+&& echo "deb http://ftp.debian.org/debian stretch-backports main" > /etc/apt/sources.list.d/stretch-backports.list \
+&& apt update \
+&& apt --target-release stretch-backports install -y golang-go \
+&& apt install -y \
+git \
+curl \
+build-essential \
+libseccomp-dev \
+btrfs-tools \
+libdevmapper-dev \
+musl-tools \
+util-linux \
+ca-certificates
+
+
+
+
+# Build firecracker itself
+# TODO can we cache cargo artifacts like we do go modules?
+FROM build-base as firecracker-build
+SHELL ["/bin/bash", "-c"]
+RUN curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain 1.32.0 \
+&& source /root/.cargo/env \
```

Comment:
Can we add tabbing here and the various other places? This will visually hint that this is grouped with the statement above.

Reply:
Sure